modAionImpl/src/org/aion/zero/impl/blockchain/AionBlockchainImpl.java

```diff
@@ -2767,4 +2751,179 @@ public void pruneOrRecoverState(boolean dropArchive, AionGenesis genesis, Logger
+                            result = tryToConnectAndFetchSummary(new BlockWrapper(block));
+                            long t2 = System.currentTimeMillis();
+                            LOG.info("<import-status: hash = " + block.getShortHash() + ", number = " + block.getNumber()
+                                    + ", txs = " + block.getTransactionsList().size() + ", result = " + result.getLeft()
+                                    + ", time elapsed = " + (t2 - t1) + " ms, td = " + getTotalDifficulty() + ">");
+                        } catch (Throwable t) {
+                            // we want to see the exception and the block where it occurred
+                            t.printStackTrace();
+                            if (t.getMessage() != null && t.getMessage().contains("Invalid Trie state, missing node ")) {
+                                LOG.info("The exception above is likely due to a pruned database and NOT a consensus problem.\n"
+                                        + "Rebuild the full state by editing the config.xml file or running ./aion.sh --state FULL.\n");
+                            }
+                            result =
+                                    new Pair<>() {
+                                        @Override
+                                        public AionBlockSummary setValue(AionBlockSummary value) {
+                                            return null;
+                                        }
+
+                                        @Override
+                                        public ImportResult getLeft() {
+                                            return ImportResult.INVALID_BLOCK;
+                                        }
+
+                                        @Override
+                                        public AionBlockSummary getRight() {
+                                            return null;
+                                        }
+                                    };
+
+                            fail = true;
+                        }
+
+                        if (!result.getLeft().isSuccessful()) {
+                            LOG.error("Consensus break at block:\n" + block);
+                            LOG.info("Import attempt returned result "
+                                    + result.getLeft()
+                                    + " with summary\n"
+                                    + result.getRight());
+
+                            if (repository.isValidRoot(repository.getBestBlock().getStateRoot())) {
+                                LOG.info("The repository state trie was:\n");
+                                LOG.info(repository.getTrieDump());
+                            }
+
+                            fail = true;
+                            break;
+                        }
+
+                        if (currentBlock % stepSize == 0) {
+                            double time = System.currentTimeMillis() - start;
```

Comment:
I prefer the calculation to use long directly. Even it has a little rounding issue.
But we can refactor it in a new task.

The estimate can be represent by TimeUnit conversion.

Reply:
added a Jira task for this and enhancing the time estimates